volatility/framework/symbols/windows/extensions/__init__.py

```diff
@@ -608,6 +616,32 @@ def get_vad_root(self):
             # windows xp and 2003
             return self.VadRoot.dereference().cast("_MMVAD")
 
+    def environment_variables(self):
+        """Generator for environment variables. 
+
+        The PEB points to our env block - a series of null-terminated
+        unicode strings. Each string cannot be more than 0x7FFF chars. 
+        End of the list is a quad-null. 
+        """
+        context = self._context
+        process_space = self.add_process_layer()
+
+        try:
+            block = self.get_peb().ProcessParameters.Environment
+            block_size = self.get_peb().ProcessParameters.EnvironmentSize
+            envars = context.layers[process_space].read(block, block_size).decode().split('\x00\x00\x00')[:-1]
```

Comment:
Could you explain why we're spliting on nulls, and why there's a decode applied to the bytes without an encoding (so `utf-8` decoding?  My guess would be that this should be `.decode('utf-16-le')` or `utf-16-be` or something, with an errors parameter (perhaps `replace`?), and then we can just split on the null character as normal?